Zend/zend_object_handlers.c

```diff
@@ -616,6 +616,8 @@ ZEND_API zval *zend_std_read_property(zend_object *zobj, zend_string *name, int
 					 * to make sure no actual modification is possible. */
 					ZVAL_COPY(rv, retval);
 					retval = rv;
+				} else if (Z_PROP_FLAG_P(retval) & IS_PROP_REINITABLE) {
+					Z_PROP_FLAG_P(retval) &= ~IS_PROP_REINITABLE;
```

Comment:
When is this necessary? If I understand correctly this would disable assignment after `$foo->bar->baz = 'baz';` (where `bar` is the `readonly` property). I guess that makes sense but should be tested.

Reply:
Huh, actually no because in this case the `if` condition above would've matched. So I don't understand what this does.